hazelcast/src/test/java/com/hazelcast/client/protocol/MessageTaskSecurityTest.java

```diff
@@ -0,0 +1,185 @@
+        skip(RemovePartitionLostListenerMessageTask.class, "Adds an internal listener");
+        skip(FetchSchemaMessageTask.class, "Fetch compact-serialization schema");
+        skip(SendAllSchemasMessageTask.class, "Send compact-serialization schemas");
+        skip(SendSchemaMessageTask.class, "Send a compact-serialization schema");
+        skip(TriggerPartitionAssignmentMessageTask.class, "Triggers first partition arrangement");
+        skip(AddCPGroupAvailabilityListenerMessageTask.class, "Listener for the cluster-topology change");
+        skip(AddCPMembershipListenerMessageTask.class, "Listener for the cluster-topology change");
+        skip(RemoveCPGroupAvailabilityListenerMessageTask.class, "Listener for the cluster-topology change");
+        skip(RemoveCPMembershipListenerMessageTask.class, "Listener for the cluster-topology change");
+        skip(CreateRaftGroupMessageTask.class, "Initial message while creating a Client proxy for any CP object");
+        skip(SqlExecuteMessageTask.class, "Permissions for specific objects are checked based on parsed query text");
+        skip(SqlCloseMessageTask.class, "Follow up SQL message where queryId is present");
+        skip(SqlFetchMessageTask.class, "Follow up SQL message where queryId is present");
+        skip(AuthenticationMessageTask.class, "Authentication message processing");
+        skip(AuthenticationCustomCredentialsMessageTask.class, "Authentication message processing");
+        skip(ExperimentalTpcAuthenticationMessageTask.class, "Authentication message processing");
+        skip(ExperimentalAuthenticationCustomCredentialsMessageTask.class, "Authentication message processing");
+        skip(CreateProxiesMessageTask.class, "Permissions handled in beforeProcess() method");
+    }
+
+    @Test
+    public void testGetRequiredPermissions() throws Exception {
+        Reflections reflections = new Reflections("com.hazelcast");
+        Set<Class<? extends AbstractMessageTask>> subTypes = reflections.getSubTypesOf(AbstractMessageTask.class);
+        for (Class clazz : subTypes) {
+            if (!Modifier.isAbstract(clazz.getModifiers())) {
+                assertGetRequiredPermission(clazz.getName());
+            }
+        }
+    }
+
+    @Test
+    public void testCreateProxiesOverridesBeforeProcess() throws Exception {
+        assertNotNull(CreateProxiesMessageTask.class.getDeclaredMethod("beforeProcess"));
+    }
+
+    private void assertGetRequiredPermission(String clsname) throws Exception {
+        if (SKIP_CLASS_MAP.containsKey(clsname)) {
+            return;
+        }
+        boolean returnsNull = doesGetRequiredPermissionSimpleReturnNull(clsname);
+        assertFalse(clsname + " returns null in getRequiredPermission()", returnsNull);
+    }
+
+    private boolean doesGetRequiredPermissionSimpleReturnNull(String clsname) throws NotFoundException, Exception, BadBytecode {
+        if (clsname == null) {
```

Comment:
how this clsName can be null 🤔 ?

Reply:
To be honest, I didn't check the behavior of the `ClassFile.getSuperclass()` call, but this seemed to me like a safe approach.